main.py

```diff
@@ -345,10 +348,10 @@ def get_board(self, access_token_in):
 
         new_img = Image.new("RGB", (new_img_width, new_img_height))
         dx_offset = 0
-        for idx, img in enumerate(imgs):
-            logger.debug("Adding image: {}", img)
+        for idx, img in enumerate(sorted(imgs, key=lambda x: x[0])):
+            logger.debug("Adding image (ID {}): {}", img[0], img[1])
             dx_offset = int(canvas_details["canvasConfigurations"][idx]["dx"])
-            new_img.paste(img, (dx_offset, 0))
```

Comment:
At this point I would question why we are using a list instead of a dict. But if it works... :)